### DISCUSSION TITLE
How to use spock to mock static Method

### DISCUSSION THREAD

**Discussion by MagicWang0907** [spockframework/spock#1757] [Q&A] [answered]
The latest version no longer exists at org.spockFramework.runtime.sputnik, so how to mock static classes without Sputnik? i wanna an example

**Comment by Vampire** [comment, marked as answer]
Nothing changed regarding mocking of static methods.
The built-in mocking of Spock does not allow it for Java mocks.
If your code you test is written in Groovy, you can use a global Groovy mock to mock static methods like documented.
If you need to mock static methods in Java code, you need to use some other facility outside Spock currently.
But in most cases it would be preferable to rewrite the code under test to not need mocking static method calls.

**Reply by MagicWang0907** [reply]
get it, thanks for your answer

**Comment by AndreasTu** [comment]
As a note I am currently working on PRs in spock-core to allow static mocking (with help of Mockito) in the future.
But this is not yet part of any Spock release. See PR #1756

**Reply by MagicWang0907** [reply]
It looks very practical, and I hope it will be released soon.
